morphia/src/main/java/org/mongodb/morphia/DatastoreImpl.java

```diff
@@ -286,30 +285,21 @@ public void ensureCaps() {
             LOG.trace("Executing findAndModify(" + dbColl.getName() + ") with delete ...");
         }
 
-        final DBObject result = dbColl.findAndModify(query.getQueryObject(), query.getFieldsObject(), query.getSortObject(), true,
-                                                     null, false, false);
+        FindAndModifyOptions copy = options.copy()
+                                           .projection(query.getFieldsObject())
+                                           .sort(query.getSortObject())
+                                           .returnNew(false)
+                                           .upsert(false)
+                                           .remove(true);
 
-        if (result != null) {
-            return mapper.fromDBObject(this, query.getEntityClass(), result, createCache());
-        }
+        final DBObject result = dbColl.findAndModify(query.getQueryObject(), copy.getOptions());
 
-        return null;
-    }
+        return mapper.fromDBObject(this, query.getEntityClass(), result, createCache());
```

Comment:
Still, this should be rolled back, because Mapper.fromDBObject logs an error when passed a null dbObject and it's not an error in this situation.